app/gui/src/dashboard/authentication/cognito.ts

```diff
@@ -184,6 +188,82 @@ interface CognitoError {
   readonly message: string
 }
 
+/**
+ * Return type for Confirm sign up endpoint
+ */
+export type ConfirmSignInReturn = Promise<
+  results.Err<AmplifyError> | results.Ok<cognito.CognitoUser>
+>
+
+/**
+ * Return type for Setup TOTP endpoint
+ */
+export interface SetupTOTPReturn {
+  /** The URL to scan the QR code */
+  readonly url: string
+  /** The secret to use for the TOTP */
+  readonly secret: string
+}
+
+/**
+ * Interface that represents Auth Provider API
+ * Currently, it's tightly coupled with Cognito, but in the future, it should be decoupled from
+ * Cognito and be able to be used with other Auth Providers.
+ *
+ * Currently used in unit tests to mock the Auth Provider API
+ */
+export interface ISessionProvider {
```

Comment:
we typically don't do `IInterfaceName` in TS

Reply:
Yeah, worth noting that we use `interface` as much as possible. And in most of the cases we don't need the `I` prefix, but for these interfaces, that supposed to be as `implementation` for `classes`, I think this might make sense, WDYT?